### DISCUSSION TITLE
Auto removal child object with morphed relation

### DISCUSSION THREAD

**Discussion by allsilaevex** [cycle/orm#360] [Q&A] [answered]
I see in the [documentation](https://cycle-orm.dev/docs/relation-has-many/2.x/en#:~:text=The%20child%20object%20will%20be%20removed%20during%20the%20persist%20operation.) that for a `HasOne` relationship the child object is deleted automatically if `nullable` as false. I have a `BelongsToMorphed` relationship, can I expect the same behavior?

**Comment by roxblnfk** [comment, marked as answer]
> Please note, the ORM will not automatically trigger the delete operation for related entities and will rely on foreign key rules set in the database.

Morphed relation can't have FK. By the reason you should delete childs automatically.

BTW I recommend you remake morphed relations to STI/JTI
